omniconf/tests/backends/test_vault.py

```diff
@@ -0,0 +1,268 @@
+#
+# This file is part of omniconf, a.k.a. python-omniconf .
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 3.0 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library. If not, see
+# <http://www.gnu.org/licenses/>.
+
+import hvac
+from hvac.tests.util import ServerManager
+from mock import patch, ANY
+from omniconf.backends import available_backends
+from omniconf.backends.vault import VaultBackend
+from omniconf.config import ConfigRegistry
+from omniconf.exceptions import InvalidBackendConfiguration
+from omniconf.setting import SettingRegistry
+from tempfile import NamedTemporaryFile
+import nose.tools
+import unittest
+
+VAULT_CONFIG = """
+backend "inmem" {
+}
+
+listener "tcp" {
+address = "127.0.0.1:18200"
+tls_disable = 1
+}
+
+disable_mlock = true
+"""
+
+NORMAL_POLICY = """
+path "secret/*" {
+    capabilities = ["read"]
+}
+"""
+
+DENY_POLICY = """
+path "secret/*" {
+    capabilities = ["deny"]
+}
+"""
+
+NORMAL_TOKEN = {
+    "id": "NORMAL",
+    "policies": ["normal"]
+}
+
+DENY_TOKEN = {
+    "id": "DENY",
+    "policies": ["deny"]
+}
+
+
+def test_vault_backend_in_available_backends():
+    nose.tools.assert_in(VaultBackend, available_backends)
+
+
+class TestVaultBackend(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        with NamedTemporaryFile() as f:
+            f.write(VAULT_CONFIG.encode('utf-8'))
+            f.flush()
+
+            cls.root_client = hvac.Client(url="http://localhost:18200")
+            cls.manager = ServerManager(config_path=f.name,
+                                        client=cls.root_client)
+            cls.manager.start()
+
+            try:
+                cls.manager.initialize()
+                cls.manager.unseal()
+                cls.root_client.token = cls.manager.root_token
+                cls.root_client.set_policy("deny", DENY_POLICY)
+                cls.root_client.set_policy("normal", NORMAL_POLICY)
+                cls.deny_token = cls.root_client.create_token(
+                    id="DENY", policies=["deny"])['auth']['client_token']
+                cls.normal_token = cls.root_client.create_token(
+                    id="NORMAL", policies=["normal"])['auth']['client_token']
+            except:  # pragma: nocover
+                cls.manager.stop()
+                raise
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.manager.stop()
+
+    def setUp(self):
+        self.assertTrue(self.root_client.is_authenticated)
```

Comment:
If the behaviour of the root_client is predictable, there should be no need for the assertion. If it is unpredictable, you should definitely use mock.